build_test.go

```diff
@@ -454,12 +457,93 @@ func testBuild(t *testing.T, context spec.G, it spec.S) {
 
 			Expect(buffer.String()).To(ContainSubstring("Some Buildpack some-version"))
 			Expect(buffer.String()).To(ContainSubstring("Resolving Dotnet Core ASPNet version"))
-			Expect(buffer.String()).To(ContainSubstring("Selected dotnet-aspnetcore version (using buildpack.yml): "))
+			Expect(buffer.String()).To(ContainSubstring("Selected dotnet-aspnetcore version (using BP_DOTNET_FRAMEWORK_VERSION): "))
 			Expect(buffer.String()).To(ContainSubstring("Reusing cached layer"))
 			Expect(buffer.String()).ToNot(ContainSubstring("Executing build process"))
 		})
 	})
 
+	context("when version-source of the selected entry is buildpack.yml", func() {
+		it.Before(func() {
+			entryResolver.ResolveCall.Returns.BuildpackPlanEntry = packit.BuildpackPlanEntry{
+				Name: "dotnet-aspnetcore",
+				Metadata: map[string]interface{}{
+					"version-source": "buildpack.yml",
+					"version":        "2.5.x",
+				},
+			}
+		})
+		it("returns a result that builds correctly", func() {
```

Comment:
I think the test is mostly here to assert that the warning is logged.

Reply:
True, in that case I'm going to get rid of some superfluous test assertions here too.

Reply:
JK you said that below. LOL